sympy/matrices/matrices.py

```diff
@@ -1035,6 +1041,10 @@ def extract(self, rowsList, colsList):
         """
         cols = self.cols
         flat_list = self._mat
+        if len(rowsList) > 0 and isinstance(rowsList[0], bool):
```

Comment:
Actually, you should probably check if all elements are a boolean, not just the first one (just to be safe). Easy: `all(isinstance(i, bool) for i in rowsList)`.